internal/pkg/source/source.go

```diff
@@ -130,10 +132,38 @@ func (a *Source) visitReferrers(referrers *[]ssa.Instruction) {
 				continue
 			}
 		}
+
+		// If the referrer is in a different block from the one we last visited,
+		// and it can't be reached from the block we are visiting, then stop visiting.
+		rb := r.Block()
+		if a.lastBlockVisited != nil && rb != a.lastBlockVisited && !a.canReach(a.lastBlockVisited, rb) {
```

Comment:
Idioms.

Reply:
Sure. The line is already very long though, and for me that makes it a little hard to read. Do you think I should extract part of the conditional?

Reply:
That would work.  You can also lean on the operators and space with newlines if you want, e.g.

```golang
if rb := r.Block(); a.lastBlockVisited != nil &&
    rb != a.lastBlockVisited &&
    !a.canReach(a.lastBlockVisited, rb) {
```

Reply:
Also, I didn't meant to sound glib in the first comment, I just know you can be prone to overlook the `if`-declaration idiom.

Reply:
It's okay.

Yeah you're right, in this case I think it was because of the line length, so thanks for suggesting splitting on the `&&`s.

I've been putting off writing a linter for this, but I guess I'll need to get to it :upside_down_face: